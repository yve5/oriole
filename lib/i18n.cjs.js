"use strict";Object.defineProperty(exports,Symbol.toStringTag,{value:"Module"});const p=require("react"),N=require("react-redux"),S=require("date-fns/locale"),m=require("date-fns");function $(e){const n=Object.create(null,{[Symbol.toStringTag]:{value:"Module"}});if(e){for(const o in e)if(o!=="default"){const r=Object.getOwnPropertyDescriptor(e,o);Object.defineProperty(n,o,r.get?r:{enumerable:!0,get:()=>e[o]})}}return n.default=e,Object.freeze(n)}const G=$(S),L={LANG:"fr",LANGUAGE:"Français",DATE_FORMAT:"dd/MM/yyyy",LOCALE:"fr","$number items per page":"$number éléments par page","$total records":"$total enregistrements",Add:"Ajouter","Ascending order":"Ordre croissant",Back:"Retour","Changes have been saved.":"Les changements ont été sauvegardés.","Choose a file":"Choisir un fichier",Columns:"Colonnes","Creation date":"Date de création","Descending order":"Ordre décroissant",Disabled:"Désactivé",Enabled:"Activé",First:"Premier","Hello World":"Bonjour le monde","Hide columns":"Cacher les colonnes",Last:"Dernier","Loading...":"Chargement...",Members:"Membres",Next:"Suivant",No:"Non","No record":"Aucun enregistrement","One record":"Un enregistrement",Previous:"Précédent",Reset:"Réinitialiser","This field is invalid.":"Ce champ est invalide.",Title:"Titre","to fill":"à remplir","Update of $update, created at $start by $guest":"Mise à jour le $update, créé le $start par $guest",Yes:"Oui"},j="I18N_INIT_LANG",O="I18N_CHANGE_LANG",C="I18N_SWITCH_LANG",I="I18N_LOCAL_USER_LANG",b=["LANG","LANGUAGE","DATE_FORMAT","LOCALE"],w=(e=localStorage.getItem(I))=>({type:j,lang:e||L.LANG}),v=e=>({type:O,lang:e}),M=()=>({type:C}),T=Object.freeze(Object.defineProperty({__proto__:null,changeLang:v,initLang:w,switchLang:M},Symbol.toStringTag,{value:"Module"})),k=({className:e="form-select",root:{i18n:{dictionaries:n}},changeLang:o})=>p.createElement("select",{className:e,"data-testid":"i18n-lang-select",onChange:({target:{value:r}})=>o(r)},n.map(({LANG:r,LANGUAGE:s})=>p.createElement("option",{key:r,value:r},s))),R=e=>({root:e}),x={...T},P=N.connect(R,x)(k),U=(e,n)=>{const o=[];let r;return Object.keys(e).forEach(s=>{r=!1,Object.keys(n).forEach(t=>{s===t&&(r=!0)}),r||o.push(s)}),Object.keys(n).forEach(s=>{r=!1,Object.keys(e).forEach(t=>{s===t&&(r=!0)}),r||o.push(s)}),o},E=(e,n="P",o="enUS")=>{const r=G[o];let s="-";if(e instanceof Date)n==="fromNow"?s=m.formatDistance(e,new Date,{addSuffix:!0,locale:r}):s=m.format(e,n,{locale:r});else if(typeof e=="string"){const t=m.parseISO(e,"y-MM-dd",new Date);t instanceof Date&&!isNaN(t)&&(n==="fromNow"?s=m.formatDistance(t,new Date,{addSuffix:!0,locale:r}):s=m.format(t,n,{locale:r}))}return s},g={LANG:"en",LANGUAGE:"English",DATE_FORMAT:"MM/dd/yyyy",LOCALE:"enUS","$number items per page":"$number items per page","$total records":"$total records",Add:"Add","Ascending order":"Ascending order",Back:"Back","Changes have been saved.":"Changes have been saved.","Choose a file":"Choose a file",Columns:"Columns","Creation date":"Creation date","Descending order":"Descending order",Disabled:"Disabled",Enabled:"Enabled",First:"First","Hello World":"Hello World","Hide columns":"Hide columns",Last:"Last","Loading...":"Loading...",Members:"Members",Next:"Next",No:"No","No record":"No record","One record":"One record",Previous:"Previous",Reset:"Reset","This field is invalid.":"This field is invalid.",Title:"Title","to fill":"to fill","Update of $update, created at $start by $guest":"Update of $update, created at $start by $guest",Yes:"Yes"},H=e=>{const n=[];return typeof e=="object"&&!Array.isArray(e)&&e!==null&&Object.keys(e).forEach(o=>{g[o]&&!b.includes(o)&&n.push(o)}),n},_=(e,n=!1)=>{const o=[];Object.keys(e).sort((t,i)=>{const a=t.toLowerCase(),c=i.toLowerCase();let l=1;return a<c?l=-1:a>c?l=1:t<i&&(l=-1),l}).map(t=>o.push(t));const r=[...o],s=b.slice(0,n?1:4);return s.forEach(t=>{const i=r.indexOf(t);if(i>-1)r.splice(i,1);else throw new Error(`${t} must be defined in dictionary`)}),[...s,...r]},F=e=>_(e,!0),D=(e,n=!1)=>{const o=[];return Object.keys(e).forEach(r=>{(!n||n&&!b.slice(1).includes(r))&&o.push(r)}),o},W=e=>D(e,!0),Y=({args:e,date:n,attrs:o,title:r,format:s,children:t,placeholder:i,root:{i18n:{lang:a,tsl:c}}})=>{let l="-";if(typeof n<"u")l=E(n,s,c("LOCALE"));else if(t&&a){const u=e||{};if(Object.keys(u).forEach(d=>{const f=u[d];typeof f=="object"&&f!==null&&"date"in f&&(f.lang=a)}),typeof t=="string")l=c(t,u);else{if(typeof t.length<"u")throw new Error("I18n can not translate several html tags in a row");if(t.type==="option"){const{value:d,children:f}=t.props;l=p.createElement("option",{value:d},c(f,u))}else{const d=o||[];if(r&&d.push("title"),i&&d.push("placeholder"),!Array.isArray(d)||Array.isArray(d)&&!d.length)throw new Error("I18n attributes are missing");const f={};d.forEach(h=>{typeof t.props[h]<"u"&&(f[h]=c(t.props[h],u))}),l=p.createElement(t.type,{...t.props,...f})}}}return l},q=e=>({root:e}),y=N.connect(q)(Y),B=({className:e,switchLang:n})=>p.createElement(y,{title:!0},p.createElement("button",{type:"button",title:"LANGUAGE",className:e,onClick:()=>n()},p.createElement(y,null,"LANG"))),z=e=>({root:e}),V={...T},J=N.connect(z,V)(B),A=(e,n,o={},r=!1)=>{let s;if(typeof n=="string"){const t=e[n];let i=!1;if(typeof o=="object"&&o!==null&&Object.keys(o).forEach(a=>{n===a&&(i=!0)}),i)s=n;else if(t)s=t;else{const a=`Not translated input: ${n}`;if(s=n,process.env.NODE_ENV!=="test"||r)console.warn(a);else throw new Error(a)}typeof o=="object"&&o!==null&&Object.keys(o).forEach(a=>{const c=o[a];if(typeof c=="object"&&c!==null&&"date"in c){const{date:l,format:u}=c;s=s.split(a).join(E(l,u,e.LOCALE))}else s=s.split(a).join(c)})}return s},K={tsl:(e,n)=>A(g,e,n),dictionaries:[g,L],lang:g.LANG},Q=(e,n=I)=>{const o=(t,i)=>{const a=t.findIndex(({LANG:c})=>c===i);return a===-1?0:a};let r={...K};if(Array.isArray(e)&&e[0]){const t=[];e.forEach(i=>{i.LANG===g.LANG?t.push({...g,...i}):i.LANG===L.LANG?t.push({...L,...i}):t.push(i)}),r={...r,dictionaries:t,tsl:(i,a)=>A(t[0],i,a)}}const s=localStorage.getItem(n);if(s){const t=r.dictionaries,i=o(t,s);r={...r,lang:s,tsl:(a,c)=>A(t[i],a,c)}}return(t=r,i={})=>{switch(i.type){case O:{const a=o(t.dictionaries,i.lang);return{...t,...i,tsl:(c,l)=>A(t.dictionaries[a],c,l)}}case C:{const c=(o(t.dictionaries,t.lang)+1)%t.dictionaries.length,u=t.dictionaries[c].LANG;return localStorage.setItem(n,u),{...t,...i,lang:u,tsl:(d,f)=>A(t.dictionaries[c],d,f)}}default:return t}}};exports.I18n=y;exports.LangSelect=P;exports.LangSwitch=J;exports.checkLocalTermOrder=F;exports.checkMissingTerms=U;exports.checkTermOrder=_;exports.convertCollection=D;exports.convertLocalCollection=W;exports.formatDate=E;exports.i18nReducer=Q;exports.trimTerms=H;
