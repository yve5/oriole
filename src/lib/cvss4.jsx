import {
  getScore,
  getVector,
  getMetricName,
  getDefaultMetrics,
  getMetricOptionName,
} from '../shared/cvss4';

export {
  getScore,
  getVector,
  getMetricName,
  getDefaultMetrics,
  getMetricOptionName,
};
