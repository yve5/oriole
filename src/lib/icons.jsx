import SvgContainer, { getIconComponent } from '../shared/icons';

export default SvgContainer;
export { getIconComponent };
