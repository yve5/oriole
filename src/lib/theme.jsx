import Theme from '../shared/theme';

export { Theme };
