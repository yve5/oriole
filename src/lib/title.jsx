import Title from '../shared/title';

export { Title };
