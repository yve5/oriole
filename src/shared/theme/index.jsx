import Theme from './components/Theme';

export default Theme;
