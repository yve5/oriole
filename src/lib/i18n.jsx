import { I18n, LangSwitch, i18nReducer } from '../shared/i18n';

export { I18n, LangSwitch, i18nReducer };
