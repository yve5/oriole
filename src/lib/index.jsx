import Theme from '../shared/theme';
import { I18n, i18nState, i18nReducer } from '../shared/i18n';

export { Theme, I18n, i18nState, i18nReducer };
