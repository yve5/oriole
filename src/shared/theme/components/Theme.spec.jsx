import React from 'react';
import { vi } from 'vitest';
import { act, fireEvent, render } from '@testing-library/react';

import { Theme } from '../../../lib/theme';

const mockMatchMedia = (matches) => ({
  removeEventListener: vi.fn(),
  addEventListener: vi.fn(),
  dispatchEvent: vi.fn(),
  matches,
});

describe('Theme', () => {
  it('renders the component. default', async () => {
    window.matchMedia = vi.fn().mockImplementation(() => mockMatchMedia(false));

    const { asFragment, getByTestId, rerender } = render(<Theme />);

    rerender(<Theme />);

    await act(async () => {
      fireEvent.click(getByTestId('theme-button'));
    });

    rerender(<Theme />);

    await act(async () => {
      fireEvent.click(getByTestId('theme-button'));
    });

    rerender(<Theme />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the component. dark', async () => {
    window.matchMedia.mockImplementation(() => mockMatchMedia(true));

    const { asFragment, getByTestId, rerender } = render(<Theme />);

    rerender(<Theme />);

    await act(async () => {
      fireEvent.click(getByTestId('theme-button'));
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
