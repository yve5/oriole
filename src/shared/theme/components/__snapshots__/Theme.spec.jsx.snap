// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Theme > renders the component. dark 1`] = `
<DocumentFragment>
  <button
    class="nav-link text-secondary text-white"
    data-testid="theme-button"
    title="light"
    type="button"
  >
    <svg
      class="svg-icon svg-icon--md"
      fill="currentColor"
      height="16"
      viewBox="0 -960 960 960"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M480-360q50 0 85-35t35-85q0-50-35-85t-85-35q-50 0-85 35t-35 85q0 50 35 85t85 35Zm0 40q-66.85 0-113.42-46.58Q320-413.15 320-480t46.58-113.42Q413.15-640 480-640t113.42 46.58Q640-546.85 640-480t-46.58 113.42Q546.85-320 480-320ZM80-460q-8.5 0-14.25-5.76T60-480.03q0-8.51 5.75-14.24T80-500h100q8.5 0 14.25 5.76t5.75 14.27q0 8.51-5.75 14.24T180-460H80Zm700 0q-8.5 0-14.25-5.76T760-480.03q0-8.51 5.75-14.24T780-500h100q8.5 0 14.25 5.76t5.75 14.27q0 8.51-5.75 14.24T880-460H780ZM479.97-760q-8.51 0-14.24-5.75T460-780v-100q0-8.5 5.76-14.25t14.27-5.75q8.51 0 14.24 5.75T500-880v100q0 8.5-5.76 14.25T479.97-760Zm0 700q-8.51 0-14.24-5.75T460-80v-100q0-8.5 5.76-14.25t14.27-5.75q8.51 0 14.24 5.75T500-180v100q0 8.5-5.76 14.25T479.97-60ZM254.46-678.77l-57.61-55.85q-5.85-5.61-5.73-13.76.11-8.16 5.96-14.77 6.38-6.62 14.15-6.62 7.77 0 14.15 6.62L282-706.31q6.38 6.62 6.38 14.16 0 7.53-6 14.15t-13.65 6.12q-7.65-.5-14.27-6.89Zm480.16 481.92L678-253.69q-6.38-6.62-6.38-14.27 0-7.66 6.38-14.04 5.23-6.62 13.08-6.12 7.84.5 14.46 6.89l57.61 55.85q5.85 5.61 5.73 13.76-.11 8.16-5.96 14.77-6.38 6.62-14.15 6.62-7.77 0-14.15-6.62ZM678-677.62q-6.62-6-6.12-13.65t6.89-14.27l55.85-57.61q5.61-5.85 13.76-5.73 8.16.11 14.77 5.96 6.62 6.38 6.62 14.15 0 7.77-6.62 14.15L706.31-678q-6.62 6.38-14.16 6.38-7.53 0-14.15-6ZM196.85-196.68q-6.62-6.78-6.62-14.55 0-7.77 6.62-14.15L253.69-282q6.62-6.38 14.27-6.38 7.66 0 14.04 6.38 5.85 5.23 5.35 13.08-.5 7.84-6.12 14.46l-55.85 57.61q-6.38 6.62-14.15 6.5-7.77-.11-14.38-6.33ZM480-480Z"
        stroke="currentColor"
        stroke-width="12"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`Theme > renders the component. default 1`] = `
<DocumentFragment>
  <button
    class="nav-link text-secondary text-white"
    data-testid="theme-button"
    title="auto"
    type="button"
  >
    <svg
      class="svg-icon svg-icon--md"
      fill="currentColor"
      height="16"
      viewBox="0 -960 960 960"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M480.13-120q-74.67 0-140.41-28.34-65.73-28.34-114.36-76.92-48.63-48.58-76.99-114.26Q120-405.19 120-479.87q0-74.67 28.34-140.41 28.34-65.73 76.92-114.36 48.58-48.63 114.26-76.99Q405.19-840 479.87-840q74.67 0 140.41 28.34 65.73 28.34 114.36 76.92 48.63 48.58 76.99 114.26Q840-554.81 840-480.13q0 74.67-28.34 140.41-28.34 65.73-76.92 114.36-48.58 48.63-114.26 76.99Q554.81-120 480.13-120ZM500-160.69q123.62-8.08 211.81-98.35T800-480q0-130.69-87.42-220.19-87.43-89.5-212.58-99.12v638.62Z"
        stroke="currentColor"
        stroke-width="12"
      />
    </svg>
  </button>
</DocumentFragment>
`;
